### PR TITLE
refactor dwjs to es6

### PR DESCRIPTION
**Pull request by gka**
Seeing all this 8-year old JS code moved into this repo was too much for me, so I had to refactor it to modern JS. Also I got rid of (most of) the code duplication by importing stuff from `@datawrapper/shared` that was already refactored.

One exception was `dw.chart` which I decided to refactor 1:1 instead of using the newer `@datawrapper/shared/chart.js`. The new version has a different API, mostly due to the fact that extends the Svelte2 store which already used `store.get` so I had to rename `chart.get` to `chart.getMetadata` etc. I don't want to refactor all places that use `chart.get` right now, so this piece of code duplication will remain for now.

So far this is working pretty well for me. There was only one [tiny adjustment](https://github.com/datawrapper/datawrapper/pull/336) I needed to make in `datawrapper`, and one more in [render.js](https://github.com/datawrapper/chart-core/pull/18/files#diff-a485abf5b8f49de7f313d7799df3faf4R140).

### Bonus goal: getting rid of jQuery

Of course, now that we're this far we could also maybe get rid of the jQuery dependency!! There are only a few places left where we still use it:

**`lib/render.js`**
- [x] some `$(el).addClass` and `$(el).removeClass`
- [x] some `$(el).height`
- [x] the GetTheData link handler `$(el).click`

**`lib/dw/chart.js`**
- [x] for `$.Callbacks` *Now replaced with a very simple [implementation](https://github.com/datawrapper/chart-core/pull/18/files#diff-27fdba1ad6b384dcb2724f5ec53b2374R5-R18)*
- [x] uses `$.getScript` to load JS. *We're now using [loadScript](https://github.com/datawrapper/shared#loadScript) instead.*
- [x] and to set some classes on the parent container div using `$(el).addClass`. *We're now using the native `Element.classList.add`, hoping the IE11 polyfill does its job!*

**`lib/dw/visualization.base.js`**
- [x] for internal event handling using `$.Deferred`
- [x] to clear the entire chart in `vis.reset()` 
- [x] and to render watermarks in `vis.postRendering`. *Now implemented in `Watermark.svelte`!*

**`lib/dw/utils/filter.js`**
- [x]  this is what renders the old UI for changing the active "tab" in the old column chart + election donut. *Now visualizations import this directly and we no longer include it in the `dw.js` bundle*

**`lib/dw/utils/getNonChartHeight.js`**
- [x]  - makes excessive use of  `$(el).outerHeight(true)` to measure the available chart height 

**`lib/dw/utils/index.js`**
- [x] one call to `$(window).height()`

### CODE REVIEW

**Comment by davidkokkelink** [comment]
very cool! about the filter UI: given that it is only used in column and election donut chart, which use jQuery anyway, we could also move `lib/dw/utils/filter.js` as is to `@datawrapper/shared` and simply import it, right? that way we reduce the bundle size for most chart types, and save ourselves the rewrite

**Comment by ilokhov** [comment]
Wow, this is great! With regards to getting rid of jQuery, it should also be straight-forward to [use vanilla JS in place of functions like `outerHeight` and `height`](http://youmightnotneedjquery.com/).

**Comment by gka** [comment]
@davidkokkelink we might even be able to import directly from `@datawrapper/chart-core/lib/dw/utils/filter`. I'll try that.

File: lib/render.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
-/* globals dw, $, Blob */
+/* globals dw, Blob */
```
Comment (sto3psl) [inline lib/render.js:1]: 🙌

**Comment by elanals** [comment]
`d3-lines` use `.data` on the `el` object ([here](https://github.com/datawrapper/plugin-d3-lines/blob/node-publishing/src/lines/main.js#L2221-L2222)) which doesn't work now that it's not a jquery object, so that needs to be refactored

**Comment by gka** [comment]
already fixed!

**Comment by elanals** [comment]
ok nice, sorry! I just wanted to also have it publicly documented just in case :)

**Pull request merged**